src/core/include/mp-units/framework/quantity_spec.h

```diff
@@ -1344,15 +1345,21 @@ template<QuantitySpec From, QuantitySpec To>
     using enum specs_convertible_result;
     return res == no ? no : yes;
   };
-  if constexpr ((NamedQuantitySpec<decltype(from_kind)> && NamedQuantitySpec<decltype(to_kind)>) ||
-                get_complexity(from_kind) == get_complexity(to_kind))
-    return convertible_impl(from_kind, to_kind);
-  else if constexpr (get_complexity(from_kind) > get_complexity(to_kind))
-    return exploded_kind_result(
-      convertible_impl(get_kind_tree_root(explode<get_complexity(to_kind)>(from_kind).quantity), to_kind));
+  if constexpr ((NamedQuantitySpec<decltype(MP_UNITS_IS_CONST_EXPR(from_kind))> &&
+                 NamedQuantitySpec<decltype(MP_UNITS_IS_CONST_EXPR(to_kind))>) ||
+                get_complexity(MP_UNITS_IS_CONST_EXPR(from_kind)) == get_complexity(MP_UNITS_IS_CONST_EXPR(to_kind)))
+    return convertible_impl(MP_UNITS_IS_CONST_EXPR(from_kind), MP_UNITS_IS_CONST_EXPR(to_kind));
```

Comment:
I do not think that the macro in the `return` expression is needed? Does the below work?

```suggestion
  if constexpr ((NamedQuantitySpec<From)> && NamedQuantitySpec<To>) || get_complexity(From{}) == get_complexity(To{}))
    return convertible_impl(from_kind, to_kind);
```

Reply:
If so, then the rest could also be simplified alike.

Reply:
You are right for this specific line, but not for the other return statements. I have to admit that I was a bit lazy and did not try to find the cases where the workaround is not needed.

Reply:
Please note that besides the return statement I also changed `constexpr` branches a lot. There are more places in the code like that to fix in a similar way.

Reply:
removed MP_UNITS_IS_CONST_EXPR